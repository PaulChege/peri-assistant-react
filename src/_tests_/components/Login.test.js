import React from "react";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import configureStore from "redux-mock-store";
import Login from "../../components/Login";
import thunk from "redux-thunk";
const mockStore = configureStore([thunk]);

describe("Login Component", () => {
  let store;
  let component;
  beforeEach(() => {
    store = mockStore({
      errors: {}
    });
    store.dispatch = jest.fn();

    component = renderer.create(
      <Provider store={store}>
        <Login />
      </Provider>
    );
  });
  it("should render form", () => {
    expect(component).toMatchSnapshot();
  });
  it("should submit form with email and password", () => {
    renderer.act(() => {
      component.root.findByType("form").props.onSubmit();
    });
    expect(store.dispatch).toHaveBeenCalledTimes(5);
  });
});
