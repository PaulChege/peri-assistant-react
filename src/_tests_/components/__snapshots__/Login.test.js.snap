// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component should render form 1`] = `
<div
  className="col-sm-4 mx-auto"
>
  <br />
  <h3>
    Log In
  </h3>
  <p
    className="text-danger"
  />
  <form
    id="loginForm"
    onSubmit={[Function]}
  >
    <div
      className="form-group"
    >
      <input
        className="form-control"
        name="email"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        placeholder="Email"
        type="email"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <input
        className="form-control"
        name="password"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        placeholder="Password"
        type="password"
        value=""
      />
    </div>
    <button
      className="btn btn-primary"
    >
      Log In
    </button>
  </form>
</div>
`;
